lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/RandomSpec.hs

```diff
@@ -151,15 +160,106 @@ spec = do
                 , validateSelection = noValidation
                 , utxoInputs = [oneAda, oneAda, oneAda, oneAda]
                 , txOutputs = 2*oneAda :| [oneAda `div` 2]
+                , totalWithdrawal = 0
                 })
 
+        coinSelectionUnitTest random "withdrawal simple"
+            (Right $ CoinSelectionResult
+                { rsInputs = [1]
+                , rsChange = []
+                , rsOutputs = [2]
+                })
+            (CoinSelectionFixture
+                { maxNumOfInputs = 100
+                , validateSelection = noValidation
+                , utxoInputs = [1]
+                , txOutputs = 2 :| []
+                , totalWithdrawal = 1
+                })
+
+        coinSelectionUnitTest random "withdrawal multi-output"
+            (Right $ CoinSelectionResult
+                { rsInputs = [1,1]
+                , rsChange = []
```

Comment:
why it is not `[1,1]` ?

Reply:
inps  + withdrawals : `[1,1,2]` to  cover [2,2] gives [] change

Reply:
These examples are indeed a bit tricky to follow, I tried to left some note above each test case to make it a bit easier.